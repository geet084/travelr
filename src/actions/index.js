
export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasErrored = (message) => ({
  type: 'HAS_ERRORED',
  message
})

export const fetchApodSuccess = (content) => ({
  type: 'FETCH_APOD_SUCCESS',
  content
})

export const fetchPlanetsSuccess = (planets) => ({
  type: 'FETCH_PLANETS_SUCCESS',
  planets
})

export const fetchImagesSuccess = (images) => ({
  type: 'FETCH_IMAGES_SUCCESS',
  images
})

export const setArrivalTime = (arrivalTime) => ({
  type: 'SET_ARRIVAL_TIME',
  arrivalTime
})

export const setBodies = (bodies) => ({
  type: 'SET_BODIES',
  bodies
})

export const setUserInfo = ({userDate, elapsedDays}) => ({
  type: 'SET_USER_INFO',
  userDate,
  elapsedDays,
})