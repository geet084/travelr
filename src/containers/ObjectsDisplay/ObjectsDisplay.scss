.display {
  color: white;
  position: relative;
  top: 170px;
  left: 34%;
  font-size: 2rem;
  border-radius: 30px;
  background-color: rgba(0,0,0,0.7);
  box-shadow: 0px 0px 5px 5px rgba(0,0,0,0.7);
}

.imgs {
  height: 100%;
  position: fixed; 
  left: 50%; 
  top: 0;
  transform: translateX(-50%);
  z-index: -1;
}