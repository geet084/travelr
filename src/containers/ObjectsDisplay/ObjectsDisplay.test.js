import React from 'react';
import { shallow } from 'enzyme';
import { ObjectsDisplay, mapStateToProps, mapDispatchToProps } from './ObjectsDisplay';
import { getImageSuccess } from '../../actions';

describe('ObjectsDisplay', () => {
  let wrapper;
  let mockProps;

  describe('initial state with NO props', () => {
    it('should match snapshot with no props', () => {
      mockProps = {
        handleImages: jest.fn(),
        info: { images: [] },
        images: { currentImage: { href: '' } },
      };
      wrapper = shallow(<ObjectsDisplay {...mockProps} />);

      expect(wrapper).toMatchSnapshot();
    })
  })

  describe('initial state with props', () => {
    beforeEach(() => {
      mockProps = {
        handleImages: jest.fn(),
        info: { images: ['image_id'] },
        images: { currentImage: { href: 'some href' } },
      };

      wrapper = shallow(<ObjectsDisplay {...mockProps} />)
    });

    it('should match snapshot with props', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should call handleImages with url when component mounts', () => {
      const expectedUrl = 'https://images-api.nasa.gov/asset/image_id';

      expect(mockProps.handleImages).toHaveBeenCalledWith(expectedUrl, getImageSuccess);
    });
  });

  describe('componentDidUpdate', () => {
    it('should not call getObjectsDisplayImage if there is an href provided', () => {
      mockProps = {
        handleImages: jest.fn(),
        info: { images: ['image_id'] },
        images: { currentImage: { href: 'some href' } },
      };

      wrapper = shallow(<ObjectsDisplay {...mockProps} />);
      wrapper.instance().getDisplayImage = jest.fn();
      wrapper.setProps({ images: { currentImage: { href: '' } } });

      expect(wrapper.instance().getDisplayImage).toHaveBeenCalledTimes(0);
    });

    it('should should call getDisplayImage if there is NO href provided (page reload)', () => {
      mockProps = {
        handleImages: jest.fn(),
        info: { images: ['image_id'] },
        images: { currentImage: { href: '' } },
      };

      wrapper = shallow(<ObjectsDisplay {...mockProps} />);
      wrapper.instance().getDisplayImage = jest.fn();

      wrapper.setProps({ images: { currentImage: { href: '' } } });

      expect(wrapper.instance().getDisplayImage).toHaveBeenCalledTimes(1);
    });
  });

  describe('getDisplayImage', () => {
    it('should get image to display if there is an image id present', () => {
      mockProps = {
        handleImages: jest.fn(),
        info: { images: ['image21'] },
        images: { currentImage: { href: '' } },
      };

      wrapper = shallow(<ObjectsDisplay {...mockProps} />);

      const expectedUrl = 'https://images-api.nasa.gov/asset/image21';

      expect(mockProps.handleImages).toHaveBeenCalledWith(expectedUrl, getImageSuccess);
    });

    it('should NOT get image to display if there is an image id present', () => {
      mockProps = {
        handleImages: jest.fn(),
        info: {},
        images: { currentImage: { href: '' } },
      };

      wrapper = shallow(<ObjectsDisplay {...mockProps} />);

      expect(mockProps.handleImages).not.toBeCalled();
    });
  });

  describe('mapDispatchToProps', () => {
    it('should mapDispatchToProps', () => {
      const mockDispatch = jest.fn();
      const mockArr = [];
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.handleImages(mockArr);
      expect(mockDispatch).toHaveBeenCalled();
    });
  });

  describe('mapStateToProps', () => {
    it('should mapStateToProps', () => {
      const mockState = {
        images: [{}],
        bodies: ['fake data'],
        planets: ['things and stuff'],
      };

      const expected = {
        images: [{}]
      };

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});