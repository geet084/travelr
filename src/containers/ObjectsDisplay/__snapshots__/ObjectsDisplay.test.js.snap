// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectsDisplay initial state with NO props should match snapshot with no props 1`] = `
<div
  className="display"
>
  <p>
    Name  - undefined
  </p>
  <p>
    Avg. Temp  - undefined
  </p>
  <p>
    Distance from the sun  - 
  </p>
  <p>
    undefined / undefined
  </p>
  <p>
    Length of day  - undefined
  </p>
  <p>
    Orbital period  - undefined
  </p>
</div>
`;

exports[`ObjectsDisplay initial state with props should match snapshot with props 1`] = `
<div
  className="display"
>
  <p>
    Name  - undefined
  </p>
  <p>
    Avg. Temp  - undefined
  </p>
  <p>
    Distance from the sun  - 
  </p>
  <p>
    undefined / undefined
  </p>
  <p>
    Length of day  - undefined
  </p>
  <p>
    Orbital period  - undefined
  </p>
  <img
    alt=""
    className="imgs"
    src="some href"
  />
</div>
`;
